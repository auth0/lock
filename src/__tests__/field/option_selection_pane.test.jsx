import React from 'react';
import { mount } from 'enzyme';

import { expectComponent, extractPropsFromWrapper, mockComponent } from 'testUtils';

jest.mock('ui/list', () => mockComponent('list'));

const getComponent = () => require('field/option_selection_pane').default;

describe('OptionSelectionPane', () => {
  const defaultProps = {
    iconUrl: 'iconUrl',
    icon: 'icon',
    items: 'items',
    name: 'option_selection_pane',
    model: {
      get: () => 'id'
    }
  };

  beforeEach(() => {
    jest.resetModules();

    jest.mock('field/actions', () => ({
      cancelOptionSelection: jest.fn(),
      selectOption: jest.fn()
    }));
  });
  it('renders correctly', () => {
    const OptionSelectionPane = getComponent();
    expectComponent(
      <OptionSelectionPane
        {...defaultProps}
        />
    ).toMatchSnapshot();
  });
  it('calls `selectOption` when selected', () => {
    let OptionSelectionPane = getComponent();

    const wrapper = mount(<OptionSelectionPane {...defaultProps} />);
    const props = extractPropsFromWrapper(wrapper);

    props.onSelect('selected');

    const {mock} = require('field/actions').selectOption;
    expect(mock.calls.length).toBe(1);
    expect(mock.calls[0]).toMatchSnapshot();
  });
  it('calls `cancelOptionSelection` when cancelled', () => {
    let OptionSelectionPane = getComponent();

    const wrapper = mount(<OptionSelectionPane {...defaultProps} />);
    const props = extractPropsFromWrapper(wrapper);

    props.onCancel();

    const {mock} = require('field/actions').cancelOptionSelection;
    expect(mock.calls.length).toBe(1);
    expect(mock.calls[0]).toMatchSnapshot();
  });
});
