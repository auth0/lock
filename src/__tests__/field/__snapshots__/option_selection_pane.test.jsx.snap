// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionSelectionPane calls \`cancelOptionSelection\` when cancelled 1`] = `
Array [
  "id",
]
`;

exports[`OptionSelectionPane calls \`selectOption\` when selected 1`] = `
Array [
  "id",
  "option_selection_pane",
  "selected",
]
`;

exports[`OptionSelectionPane renders correctly 1`] = `
<div
  data-__type="list"
  data-icon="icon"
  data-iconUrl="iconUrl"
  data-items="items"
  data-model={
    Object {
      "get": [Function],
    }
  }
  data-onCancel={[Function]}
  data-onSelect={[Function]}
/>
`;
